### PR TITLE
34 add lat/lon to EPC dataset according to UPRN

### LINKED ISSUE #34: Add lat/long to EPC dataset
Add lat / long to EPC dataset by joining [OS Open UPRN - Lat/lon dataset](https://osdatahub.os.uk/downloads/open/OpenUPRN) on UPRN.

### PR DESCRIPTION
**Pull request by crispy-wonton**
Fixes #34 

# Description

Add lat/long coordinates to each UPRN in EPC where available from [OS Open UPRN](https://osdatahub.os.uk/downloads/open/OpenUPRN). Changes:

- Add `base_getter` function to load a CSV from a ZIP file on S3. Also refactor a base_getter function name.
- Add function to `get_datasets.py` to get raw OS Open UPRN dataset
- Add `prepare_features/lat_lon.py` file to transform OS Open UPRN file in preparation for join with EPC
- Add functions to `run_scripts/add_features.py` to join lat/lon to EPC
- add OS Open UPRN source to config and README

# Instructions for Reviewer

Please could you:
- Check for general bugs/ mistakes
- Check the code works and produces the expected output with the code below

To test the code you can run the following line in terminal on the 10k EPC test set and check the outputs:
`python asf_heat_pump_suitability/pipeline/run_scripts/add_features.py --epc_path s3://asf-heat-pump-suitability/outputs/2023_Q2_EPC_enhanced_weights_sample_10k.parquet --save_output s3://asf-heat-pump-suitability/outputs/test.parquet`
 
# Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

### CODE REVIEW

File: asf_heat_pump_suitability/pipeline/run_scripts/add_features.py [hunk 1]
```diff
@@ -34,6 +37,7 @@ def main(epc_path: str, save_output: Optional[str] = None) -> pl.DataFrame:
     """
     Enhance EPC dataset with additional features:
     - mean average garden size per MSOA
+    - lat/lon per UPRN
```
Comment (lizgzil) [inline asf_heat_pump_suitability/pipeline/run_scripts/add_features.py:40]: nice to see how this script will continue to get added to with features!

File: asf_heat_pump_suitability/pipeline/run_scripts/add_features.py [hunk 2]
```diff
@@ -56,6 +62,15 @@ def main(epc_path: str, save_output: Optional[str] = None) -> pl.DataFrame:
...
     )
 
+    # Add feature: lat/long
+    logging.info("Adding lat/lon data to EPC")
+    uprn_latlon_df = lat_lon.transform_df_osopen_uprn_latlon()
+    epc_latlon_df = epc_df.select(["UPRN"])
+    epc_latlon_df = epc_latlon_df.join(uprn_latlon_df, how="left", on="UPRN")
+
+    # Join enhanced datasets together
+    enhanced_epc_df = enhanced_epc_df.join(epc_latlon_df, how="left", on="UPRN")
```
Comment (lizgzil) [inline asf_heat_pump_suitability/pipeline/run_scripts/add_features.py]: can this be done with one line?:
```
enhanced_epc_df = enhanced_epc_df.join(uprn_latlon_df, how="left", on="UPRN")
```

**Review by lizgzil — APPROVED** [review]
looks great!

Code worked, and the lat/long values are appropriate for the UK's range.

**Comment by lizgzil** [comment]
@crispy-wonton can this be merged now?

File: asf_heat_pump_suitability/pipeline/run_scripts/add_features.py (hunk 2 above)
Comment (crispy-wonton) [reply asf_heat_pump_suitability/pipeline/run_scripts/add_features.py]: Yes, not sure why I did it separately.

**Pull request merged**